src/snapred/backend/recipe/algorithm/FitMultiplePeaksAlgorithm.py

```diff
@@ -0,0 +1,98 @@
+import json
+
+import numpy as np
+from mantid.api import AlgorithmFactory, PythonAlgorithm, WorkspaceGroup, mtd
+from mantid.kernel import Direction
+from mantid.simpleapi import DeleteWorkspace
+
+from snapred.backend.dao.FitMultiplePeaksIngredients import FitMultiplePeaksIngredients
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+from snapred.backend.recipe.algorithm.PurgeOverlappingPeaksAlgorithm import PurgeOverlappingPeaksAlgorithm
+
+name = "FitMultiplePeaksAlgorithm"
+
+
+class FitMultiplePeaksAlgorithm(PythonAlgorithm):
+    def PyInit(self):
+        # declare properties
+        self.declareProperty("FitMultiplePeaksIngredients", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspaceGroup", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def PyExec(self):
+        fitPeakIngredients = FitMultiplePeaksIngredients(
+            **json.loads(self.getProperty("FitMultiplePeaksIngredients").value)
+        )
+        wsName = fitPeakIngredients.InputWorkspace
+        instrumentState = fitPeakIngredients.InstrumentState
+        crystalInfo = fitPeakIngredients.CrystalInfo
+        peakType = fitPeakIngredients.PeakType
+
+        beta_0 = instrumentState.gsasParameters.beta[0]
+        beta_1 = instrumentState.gsasParameters.beta[1]
+        FWHMMultiplierLeft = instrumentState.fwhmMultiplierLimit.minimum
+        FWHMMultiplierRight = instrumentState.fwhmMultiplierLimit.maximum
+        L = instrumentState.instrumentConfig.L1 + instrumentState.instrumentConfig.L2
+        ws = mtd[wsName]
```

Comment:
`ws = self.mantidSnapper.mtd[wsName]`
should work with latest

Reply:
Thanks, I made the requested change.